### PR TITLE
Fix a few recent (STM32F1 DOGM, SPI) warnings

### PR DESCRIPTION
**Pull request by tpruvot**
Fix a few recent warnings

### CODE REVIEW

File: Marlin/src/HAL/HAL_STM32F1/HAL_spi_STM32F1.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ void spiInit(uint8_t spiRate) {
  * @details
  */
 uint8_t spiRec(void) {
-  uint8_t returnByte = SPI.transfer(0xFF);
+  uint8_t returnByte = SPI.transfer(ff);
```
Comment (thinkyhead) [inline Marlin/src/HAL/HAL_STM32F1/HAL_spi_STM32F1.cpp:107]: What does `ff` refer to?

File: Marlin/src/lcd/dogm/HAL_LCD_com_defines.h [hunk 2]
```diff
@@ -25,52 +25,48 @@
...
+    #define U8G_COM_HAL_SW_SPI_FN  u8g_com_HAL_AVR_sw_sp_fn
     #define U8G_COM_HAL_HW_SPI_FN u8g_com_arduino_hw_spi_fn
-    uint8_t u8g_com_arduino_st7920_spi_fn(u8g_t *u8g, uint8_t msg, uint8_t arg_val, void *arg_ptr);
     #define U8G_COM_ST7920_HAL_SW_SPI u8g_com_arduino_st7920_spi_fn
+    #define U8G_COM_ST7920_HAL_HW_SPI u8g_com_arduino_st7920_hw_spi_fn
+  #else
+    #define U8G_COM_HAL_SW_SPI_FN u8g_com_arduino_std_sw_spi_fn
+    #define U8G_COM_HAL_HW_SPI_FN u8g_com_arduino_hw_spi_fn
+    #define U8G_COM_ST7920_HAL_SW_SPI u8g_com_arduino_st7920_spi_fn
+    #define U8G_COM_ST7920_HAL_HW_SPI u8g_com_arduino_st7920_hw_spi_fn
```
Comment (thinkyhead) [inline Marlin/src/lcd/dogm/HAL_LCD_com_defines.h]: I don’t think you can define these with two different values at the same time.

**Comment by thinkyhead** [comment]
Looks good, but one question, above…

File: Marlin/src/HAL/HAL_STM32F1/HAL_spi_STM32F1.cpp (hunk 1 above)
Comment (tpruvot) [reply Marlin/src/HAL/HAL_STM32F1/HAL_spi_STM32F1.cpp:107]: its a static const in framework, = 0xff but when unused makes a warning (see travis)

Comment (thinkyhead) [reply Marlin/src/HAL/HAL_STM32F1/HAL_spi_STM32F1.cpp:107]: I see. So, it's of the exact type that `SPI.transfer` is expecting.

**Pull request merged**